crates/executor/src/executor.rs

```diff
@@ -17,7 +18,157 @@ use revm::{
+        }
+    }
+
+    fn init_block_env(&mut self, header: &Header) {
+        let spec_id = revm_spec(self.chain_spec, header.number);
+        self.evm.env.cfg.chain_id = U256::from(self.chain_spec.chain().id());
+        self.evm.env.cfg.spec_id = spec_id;
+        self.evm.env.cfg.perf_all_precompiles_have_balance = false;
+        self.evm.env.cfg.perf_analyse_created_bytecodes = AnalysisKind::Raw;
+
+        revm_wrap::fill_block_env(&mut self.evm.env.block, header, spec_id >= SpecId::MERGE);
+    }
+}
+
+impl<'a, DB> BlockExecutor for Executor<'a, DB>
+where
+    DB: StateProvider,
+{
+    fn execute(
+        &mut self,
+        block: &Block,
+        signers: Option<Vec<Address>>,
+    ) -> Result<ExecutionResult, Error> {
+        let Block { header, body, ommers } = block;
+        let signers = self.recover_signers(body, signers)?;
+
+        self.init_block_env(header);
+
+        let mut cumulative_gas_used = 0;
+        // output of verification
+        let mut changesets = Vec::with_capacity(body.len());
+
+        for (transaction, signer) in body.iter().zip(signers.into_iter()) {
+            // The sum of the transaction’s gas limit, Tg, and the gas utilised in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                return Err(Error::TransactionGasLimitMoreThenAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                })
+            }
+
+            // Fill revm structure.
+            revm_wrap::fill_tx_env(&mut self.evm.env.tx, transaction, signer);
+
+            // Execute transaction.
+            let out = self.evm.transact();
+
+            // Useful for debugging
+            // let out = evm.inspect(revm::inspectors::CustomPrintTracer::default());
+            // tracing::trace!(target:"evm","Executing transaction {:?}, \n:{out:?}: {:?}
+            // \nENV:{:?}",transaction.hash(),transaction,evm.env);
+
+            let (revm::ExecutionResult { exit_reason, gas_used, logs, .. }, state) = out;
+
+            // Fatal internal error.
+            if exit_reason == revm::Return::FatalExternalError {
+                return Err(Error::ExecutionFatalError)
+            }
+
+            // Success flag was added in `EIP-658: Embedding transaction status code in receipts`.
+            // TODO for verification (exit_reason): some error should return EVM error as the block
+            // with that transaction can have consensus error that would make block
+            // invalid.
+            let is_success = match exit_reason {
+                revm::return_ok!() => true,
+                revm::return_revert!() => false,
+                _ => false,
+                //e => return Err(Error::EVMError { error_code: e as u32 }),
```

Comment:
Should we handle this error separately? Otherwise rm

Reply:
This commented line also already existed in `execute()`. So I need more feedback before removing it.

Reply:
Related to this: https://github.com/paradigmxyz/reth/issues/463
It needs upgrade to revm v3.0.0, this can be done in separate PR to not ple a lot of changes here.

Reply:
@onbjerg We need to keep this commented line. Could we mark it as resolved?